apistar/codecs/formdata.py

```diff
@@ -0,0 +1,34 @@
+from io import BytesIO
+from werkzeug.datastructures import ImmutableMultiDict
+from werkzeug.formparser import FormDataParser
+from werkzeug.http import parse_options_header
+from werkzeug.urls import url_decode
+
+from apistar.codecs.base import BaseCodec
+
+
+class MultiPartCodec(BaseCodec):
+    media_type = 'multipart/form-data'
+
+    def decode(self, bytestring, headers, **options):
+        try:
+            content_length = max(0, int(headers['content-length']))
+        except (KeyError, ValueError, TypeError):
+            content_length = None
+
+        try:
+            mime_type, mime_options = parse_options_header(headers['content-type'])
+        except KeyError:
+            mime_type, mime_options = '', {}
+
+        body_file = BytesIO(bytestring)
```

Comment:
This should be somewhat of a stopgap since this codec should have access to a file-like object rather than the framework having to read the whole request body into memory, but I couldn't figure out a quick way to get to the underlying file object. Let me know if/how I should change this.

Reply:
You could take inspiration from how `flask-api` does it:

https://github.com/flask-api/flask-api/blob/376815deca2c6ec827633e9e152e7452de1540dd/flask_api/parsers.py#L33-L57

Reply:
I would, but the problem is getting the `stream` (body file) parameter into `decode(...)` in the first place. [RequestDataComponent](https://github.com/encode/apistar/blob/3aed0e4b90c6c949b2a915b8e750658d87b0f7d5/apistar/server/validation.py#L14) depends on `Body` and I believe that represents the full request body as a bytestring. There doesn't seem to be a `BodyFile` or similar component so I figured I'd ask before going deeper. :D